sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -595,8 +601,71 @@ class Analyzer(
       }
     }
 
-    // This require transformUp to replace grouping()/grouping_id() in resolved Filter/Sort
-    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperatorsUp {
+    private def tryResolveHavingCondition(
+        a: UnresolvedHaving, agg: LogicalPlan): LogicalPlan = {
```

Comment:
actually, we just need one parameter `UnresolvedHaving`, and `agg` is just `a.child`

Reply:
Copy that, done in 1de0c75.